types/react-leaflet/react-leaflet-tests.tsx

```diff
@@ -211,30 +243,34 @@ export class DraggableExample extends Component<any, any> {
 
     toggleDraggable = () => {
         this.setState({ draggable: !this.state.draggable });
-    }
+    };
 
     updatePosition = () => {
-        const { lat, lng } = (this.refs['marker'] as MarkerInstance).leafletElement.getLatLng();
+        const {
+            lat,
+            lng,
+        } = (this.refs.marker as Marker<MarkerProps, Leaflet.Marker>).leafletElement.getLatLng();
```

Comment:
I introduced type parameters to get better type checking on extensions to react-leaflet classes. They allow extending classes to pass custom prop types and custom leafletElement types to inherited members. Unfortunately, this introduces a bit of extra syntax in cases like the one referenced above. It would be possible to reintroduce a `MarkerInstance` type that simply uses the default type parameters `MarkerProps` and `Leaflet.Marker`. I see two problems with this, though:
1. If we create `MapInstance` and `MarkerInstance` types, as in the present version, why stop there? For consistency, there should ideally be such -`Instance` types for _every_ react-leaflet class.
2. Because I've defined `Map`, `Marker`, and indeed every react-leaflet component as a true TypeScript class rather than a `const` representing the constructor, defining separate types for their instances seems unnecessary, since the type of the instance is the same as the type of the class.